controller/protocols.py

```diff
@@ -773,3 +771,149 @@ async def indy_present_proof_v2(
+            "cred_rev_id": cred_rev_id,
+            "publish": publish,
+            "notify": notify,
+        },
+        response=None,
+    )
+
+
+async def indy_anoncreds_revoke(
+    issuer: Controller,
+    holder_connection_id: Optional[str] = None,
+    v10_credential_exchange_object: Optional[V10CredentialExchange] = None,
+    v20_credential_exchange_object: Optional[V20CredExRecordDetail] = None,
+    publish: bool = False,
+    notify: bool = True,
+):
+    """Revoking an Indy credential using revocation revoke.
+    V1.0: V10CredentialExchange
+    V2.0: VV20CredExRecordDetail
+    """
+    # Passes in V10CredentialExchange
+    if v10_credential_exchange_object:
+        await post_method(
+            issuer,
+            url="/revocation/revoke",
+            rev_reg_id=v10_credential_exchange_object.revoc_reg_id,
+            cred_rev_id=v10_credential_exchange_object.revocation_id,
+            publish=publish,
+            holder_connection_id=holder_connection_id,
+            notify=notify,
+        )
+
+    # Passes in V20CredExRecordDetail
+    elif v20_credential_exchange_object and v20_credential_exchange_object.indy:
+        await post_method(
+            issuer,
+            url="/revocation/revoke",
+            rev_reg_id=v20_credential_exchange_object.indy.rev_reg_id,
+            cred_rev_id=v20_credential_exchange_object.indy.cred_rev_id,
+            publish=publish,
+            holder_connection_id=holder_connection_id,
+            notify=notify,
+        )
+    else:
+        raise ValueError(
+            "If using V1.0, try passing in a V10CredentialExchange object. If using V2.0, try passing in a V20CredExRecordDetail object."
+        )
+
+
+async def indy_anoncreds_publish_revocation(
+    issuer: Controller,
+    v10_credential_exchange_object: Optional[V10CredentialExchange] = None,
+    v20_credential_exchange_object: Optional[V20CredExRecordDetail] = None,
+    publish: bool = False,
+    notify: bool = True,
+):
+    if v10_credential_exchange_object:
+        await post_method(
+            issuer,
+            url="/revocation/publish-revocations",
+            rev_reg_id=v10_credential_exchange_object.revoc_reg_id,
+            cred_rev_id=v10_credential_exchange_object.revocation_id,
+            publish=publish,
+            notify=notify,
+        )
+
+    elif v20_credential_exchange_object and v20_credential_exchange_object.indy:
+        await post_method(
+            issuer,
+            url="/revocation/publish-revocations",
+            rev_reg_id=v20_credential_exchange_object.indy.rev_reg_id,
+            cred_rev_id=v20_credential_exchange_object.indy.cred_rev_id,
+            publish=publish,
+            notify=notify,
+        )
+    else:
+        raise ValueError(
+            "If using V1.0, try passing in a V10CredentialExchange object. If using V2.0, try passing in a V20CredExRecordDetail object."
+        )
+
+
+async def indy_anoncreds_proof_revocations(
```

Comment:
We should use the `indy_present_proof_vX` which is more generalized than this. We should eliminate this method in favor of those.